### PR TITLE
1.18 protocol

### PR DESCRIPTION
**Pull request by nickelpro**
This includes two "Add 1.18 protocol" commits which should be squashed, and are provided purely so that the difference between 1.17 and 1.18 can be easily diffed.

Also includes a fix for 1.17 vibration particles that was discovered while investigating particle changes in 1.18

### CODE REVIEW

File: data/pc/1.18/protocol.json [hunk 1]
```diff
@@ -4274,6 +4360,15 @@
             }
           ]
         ],
+        "packet_simulation_distance": [
+          "container",
+          [
+            {
+              "name": "dist",
```
Comment (rom1504) [inline data/pc/1.18/protocol.json:4367]: why not simulationDistance ?

**Review by rom1504 — CHANGES_REQUESTED** [review]
https://wiki.vg/index.php?title=Pre-release_protocol&oldid=17160

missing simulation distance in join game
type field in block entity data

File: data/pc/1.18/protocol.json (hunk 1 above)
Comment (nickelpro) [reply data/pc/1.18/protocol.json:4367]: Seems redundant and long winded, the packet gives that context

```js
client.on("simulation_distance", (packet) => {
  console.log(packet.dist);
})
```

Comment (rom1504) [reply data/pc/1.18/protocol.json:4367]: ok but let's at least have a full word: distance

**Pull request merged**